src/FSharp.Core/QueryExtensions.fs

```diff
@@ -17,268 +17,368 @@ open System.Linq.Expressions
+           typedefof<_ * _ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _, _>>
+           typedefof<_ * _ * _ * _ * _ * _ * _ * _>, typedefof<AnonymousObject<_, _, _, _, _, _, _, _>> |]
+
     let anonObjectTypes = tupleTypes |> Array.map snd
-    let tupleToAnonTypeMap = 
-        let t = new Dictionary<Type,Type>()
-        for (k,v) in tupleTypes do t.[k] <- v
+
+    let tupleToAnonTypeMap =
+        let t = new Dictionary<Type, Type>()
+
+        for (k, v) in tupleTypes do
+            t.[k] <- v
+
         t
 
-    let anonToTupleTypeMap = 
-        let t = new Dictionary<Type,Type>()
-        for (k,v) in tupleTypes do t.[v] <- k
+    let anonToTupleTypeMap =
+        let t = new Dictionary<Type, Type>()
+
+        for (k, v) in tupleTypes do
+            t.[v] <- k
+
         t
 
 
     /// Recognize anonymous type construction written using 'new AnonymousObject(<e1>, <e2>, ...)'
-    let (|NewAnonymousObject|_|) e = 
+    let (|NewAnonymousObject|_|) e =
         match e with
-        | Patterns.NewObject(ctor,args) when 
-                 let dty = ctor.DeclaringType 
-                 dty.IsGenericType && anonToTupleTypeMap.ContainsKey (dty.GetGenericTypeDefinition()) -> 
-             Some (ctor, args)
+        | Patterns.NewObject (ctor, args) when
+            let dty = ctor.DeclaringType
+
+            dty.IsGenericType
+            && anonToTupleTypeMap.ContainsKey(dty.GetGenericTypeDefinition())
+            ->
+            Some(ctor, args)
         | _ -> None
 
-    let OneNewAnonymousObject (args:Expr list) =
+    let OneNewAnonymousObject (args: Expr list) =
         // Will fit into a single tuple type
         let typ = anonObjectTypes.[args.Length - 1]
         let typ = typ.MakeGenericType [| for a in args -> a.Type |]
         let ctor = typ.GetConstructors().[0]
-        let res = Expr.NewObject (ctor, args)
-        assert (match res with NewAnonymousObject _ -> true | _ -> false)
+        let res = Expr.NewObject(ctor, args)
+
+        assert
+            (match res with
+             | NewAnonymousObject _ -> true
+             | _ -> false)
+
         res
 
-    let rec NewAnonymousObject (args:Expr list) : Expr = 
-        match args with 
+    let rec NewAnonymousObject (args: Expr list) : Expr =
+        match args with
         | x1 :: x2 :: x3 :: x4 :: x5 :: x6 :: x7 :: x8 :: tail ->
             // Too long to fit single tuple - nested tuple after first 7
-            OneNewAnonymousObject [ x1; x2; x3; x4; x5; x6; x7; NewAnonymousObject (x8 :: tail) ]
-        | args -> 
-            OneNewAnonymousObject args
-
-    let AnonymousObjectGet (e:Expr,i:int) = 
-        // Recursively generate tuple get 
+            OneNewAnonymousObject
+                [ x1
+                  x2
+                  x3
+                  x4
+                  x5
+                  x6
+                  x7
+                  NewAnonymousObject(x8 :: tail) ]
```

Comment:
There are some fantomas settings to stop this sort of thing happening if we want to use them `fsharp_max_array_or_list_number_of_items` and `fsharp_max_array_or_list_width`

Reply:
Side note, [fsharp_max_array_or_list_number_of_items](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#fsharp_max_array_or_list_number_of_items) doesn't do anything unless `fsharp_array_or_list_multiline_formatter=number_of_items`.